### DISCUSSION TITLE
Config lib

### DISCUSSION THREAD

**Discussion by mrguamos** [jooby-project/jooby#3007] [General]
seems typesafe config hasn't been touched for more than a year. maybe it's time to use other libs like smallrye config?

**Comment by SentryMan** [comment]
I'm more of an avaje-config guy myself, so I think it would be good if we could generalize the configuration to make it a CYOA and not tie it to any one implementation.

**Comment by agentgt** [comment]
Just for reference this was discussed earlier here: https://github.com/jooby-project/jooby/issues/1712#issuecomment-699657483

(scroll for responses).

The issue is as I believe @jknack expressed is that all the optional modules expect Lightbend `Config` both from a Java interface contract but also perhaps more importantly documented configuration format.

I hate to denigrate Microprofile but tried my best to be involved with Microprofile Config in hopes it could fill the niche for a config facade but that community seems to be very closed off (I am actually a little jaded about that and reminds me of old school design by committee that new Microprofile JEE was supposed to resolve...). [BTW Microprofile Config is still not a Java module](https://github.com/eclipse/microprofile-config/issues/768) so it is for sure off the table.

That being said I'm still in favor of a config abstraction for pluggable config.
